src/main/java/pixelperfect/Spaceship.java

```diff
@@ -73,12 +75,24 @@ public boolean isDead() {
   }
 
   /**
+   * Checks if the ship has reached the end of the Route.
+   * 
+   * @return A Boolean pointing out Victory.
+   */
+  public boolean isVictorious() {
+    return victorious;
+  }
+
+  /**
    * Update the Spaceship and related model classes for the step in the game loop.
    * 
    * @param tpf
    *          Time per frame.
    */
   public void update(float tpf) {
+    if (route.isCompleted(timer)) {
+      this.victorious = true;
+    }
```

Comment:
it doesnt really matter since update will happen pretty much constantly, but updating the timer should be done before isCompleted